proarc-common/src/test/java/cz/cas/lib/proarc/common/object/emods/BdmArticleMapperTest.java

```diff
@@ -0,0 +1,91 @@
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package cz.cas.lib.proarc.common.object.emods;
+
+import cz.cas.lib.proarc.common.mods.ModsUtils;
+import cz.cas.lib.proarc.common.mods.ndk.NdkMapper.Context;
+import cz.cas.lib.proarc.mods.ModsDefinition;
+import cz.cas.lib.proarc.mods.PhysicalDescriptionDefinition;
+import java.io.StringReader;
+import java.util.List;
+import javax.xml.transform.stream.StreamSource;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import static org.junit.Assert.*;
+
+/**
+ *
+ * @author Lukas Sykora
+ */
+public class BdmArticleMapperTest {
+
+    public BdmArticleMapperTest() {
+    }
+
+    @BeforeClass
+    public static void setUpClass() {
+    }
+
+    @AfterClass
+    public static void tearDownClass() {
+    }
+
+    @Before
+    public void setUp() {
+    }
+
+    @After
+    public void tearDown() {
+    }
+
+    @Test
+    public void testCreateMods() {
+        String xml = "<mods version='3.5' xmlns='http://www.loc.gov/mods/v3'>"
+                + "<physicalDescription> <form>bez média</form> </physicalDescription>"
+                + "<physicalDescription> <form>bez média</form> </physicalDescription>"
+                + "<physicalDescription> <form>počítač</form> </physicalDescription>"
+                + "<physicalDescription> <form>jiný</form> </physicalDescription>"
+                + "<physicalDescription> <form>jiný</form> </physicalDescription>"
+                + "<physicalDescription> <form>audio</form> </physicalDescription>"
+                + "</mods>";
+        ModsDefinition mods = ModsUtils.unmarshalModsType(new StreamSource(new StringReader(xml)));
+        BdmArticleMapper mapper = new BdmArticleMapper();
+        Context ctx = new Context("uuid:testId");
+        mapper.createMods(mods, ctx);
+
+        int rdamediaSum = 0;
+        int rdacarrierSum = 0;
+        int otherSum = 0;
+        for (PhysicalDescriptionDefinition pd : mods.getPhysicalDescription()) {
+            String authority = pd.getForm().get(0).getAuthority();
+            if ("rdamedia".equals(authority)) {
+                rdamediaSum++;
+            } else if ("rdacarrier".equals(authority)) {
+                rdacarrierSum++;
+            } else {
+                otherSum++;
+            }
+        }
+        assertEquals(9, mods.getPhysicalDescription().size());
+        assertEquals(5, rdamediaSum);
+        assertEquals(3, rdacarrierSum);
+        assertEquals(1, otherSum);
```

Comment:
Asi by stálo za to do komentáře napsat původ čísel assert-u.

Reply:
Okomentováno

Reply:
Špatně jsme se pochopili, myslel jsem hodnoty "9", "5", "3", "1", ne proměnných.

Reply:
Asi nechápu, jak by sis to představoval. K devítce mám napsat, že to je size celého listu. K 5, že to je počet elementů, jejichž authority = rdamedia atd? Jestli ne, tak jak jinak by sis to představoval?

Reply:
Imho to stačí jenom trochu ozřejmit, tzn. napsat, že v původním listu (v tom zdrojovém XML) je těch položek několik, ale došlo k nějakému obohacení a proto je jich teď více (případně připsat které se přidaly)

Reply:
Nene, spíš nastínit to "pomnožení" elementů, protože na vstupu je xml s 2x bez media, 1x počítač, 2x jiný, 1x audio a z toho nevidím přirozený matematický vzah k těm kontrolovaným hodnotám (9,5,3,1). 

Představoval bych si teda něco ve smyslu (příklad který neodpovídá realitě):

5: |"bez media"| * 2 + 1
3: |"jiný"| + 1

Reply:
jasný, už tě chápu